test/validation/api/ipsec/ipsec_test_in.c

```diff
@@ -1597,6 +1597,42 @@ static void test_in_ipv6_esp_udp_null_sha256_lookup(void)
 	ipsec_sa_destroy(sa);
 }
 
+static void test_ipsec_print(void)
+{
+	odp_ipsec_print();
+}
+
+static void test_ipsec_sa_print(void)
+{
+	odp_ipsec_tunnel_param_t in_tunnel;
+	odp_ipsec_sa_param_t param_in;
+	odp_ipsec_capability_t capa;
+	uint32_t window_sz = 32;
+	odp_ipsec_sa_t in_sa;
+
+	CU_ASSERT(odp_ipsec_capability(&capa) == 0);
+
+	memset(&in_tunnel, 0, sizeof(odp_ipsec_tunnel_param_t));
+
+	odp_ipsec_print();
+
+	ipsec_sa_param_fill(&param_in,
+			    true, false, 0x1234, &in_tunnel,
+			    ODP_CIPHER_ALG_AES_GCM, &key_mcgrew_gcm_3,
+			    ODP_AUTH_ALG_AES_GCM, NULL,
+			    &key_mcgrew_gcm_salt_3, NULL);
+
+	param_in.inbound.antireplay_ws = window_sz;
+
+	in_sa = odp_ipsec_sa_create(&param_in);
+
+	CU_ASSERT_NOT_EQUAL_FATAL(ODP_IPSEC_SA_INVALID, in_sa);
+
+	odp_ipsec_sa_print(in_sa);
+
+	odp_ipsec_sa_destroy(in_sa);
```

Comment:
odp_ipsec_sa_disable(sa) should be called before sa_destroy(). May be missing this call caused lots of CI jobs to fail.

Reply:
Thank you for pointing this out. In V4, I replaced the direct odp_ipsec_sa_destroy() call with ipsec_sa_destroy() function which takes care of the API sequence.